lib/engine/game/g_18_oe/step/buy_sell_par_shares.rb

```diff
@@ -1,38 +1,102 @@
 # frozen_string_literal: true
 
 require_relative '../../../step/buy_sell_par_shares'
+require_relative '../../../action/convert'
 
 module Engine
   module Game
     module G18OE
       module Step
         class BuySellParShares < Engine::Step::BuySellParShares
           def actions(entity)
+            return corporation_actions(entity) if entity.corporation?
             return [] unless entity == current_entity
+            return [] if @round.current_actions.any? { |x| x.instance_of?(Engine::Action::Par) }
             return ['sell_shares'] if must_sell?(entity)
 
             actions = []
             actions << 'buy_shares' if can_buy_any?(entity)
             actions << 'par' if can_ipo_any?(entity) || can_float_minor?(entity)
             actions << 'buy_company' if !purchasable_companies(entity).empty? || !buyable_bank_owned_companies(entity).empty?
             actions << 'sell_shares' if can_sell_any?(entity)
+            actions << 'convert' if can_convert_any?(entity)
 
-            actions << 'pass' if !can_float_minor?(entity) && !bought? && !actions.empty?
+            actions << 'pass' if !can_float_minor?(entity) && !actions.empty?
             actions
           end
 
+          def corporation_actions(entity)
+            if @round.current_actions[0].instance_of?(Engine::Action::BuyShares) &&
+              @round.current_actions[0].bundle.corporation != entity
```

Comment:
Why actions[0]? Can you not sell first?

Reply:
So the conversion in 18OE allows a single buy either before or after the conversion and is all one step. So you can't sell, buy, then convert.